src/mon/MDSMonitor.cc

```diff
@@ -985,15 +986,18 @@ bool MDSMonitor::preprocess_command(MonOpRequestRef op)
     }
     set<mds_gid_t> stopping;
     for (auto& id : ids) {
-      ostringstream ess;
       mds_gid_t gid = gid_from_arg(fsmap, id, ess);
       if (gid == MDS_GID_NONE) {
-	// the mds doesn't exist, but no file systems are unhappy, so losing it
-	// can't have any effect.
-	continue;
+        invalid_ids.push_back(std::string(id));
```

Comment:
`invalid_ids` are only used as a flag that an error has happened. You can set `r = -ENOENT` here and then check for it in the `if` below to avoid unnecessary vector operations